src/ring_theory/ideal/basic.lean

```diff
@@ -474,6 +474,20 @@ end pi
 
 end ideal
 
+--Move this somewhere better
```

Comment:
It seems to be pretty buggy if I do that. I tried to fix it by using `@one_ne_zero α _ _`, but I ended up with `h : 1 = 0` and `⊢ 1 = 0` all as elements of `α`. I think the problem was that `0` in the goal ended up being a bit different in `h`, `add_monoid.to_has_zero` and `mul_zero_class.to_has_zero`, and generally, it seems the type inferencer is not happy with either line. Oddly enough, it's fine where it is, but moving it up inside ideal just breaks it (although this might just be my inexperience showing).

Reply:
The problem was an instance conflict: the type `α` already had a declared `comm_ring` instance, so adding an `integral_domain` instance as well will give it two incompatible instances of `0`, `1`, `+`, `*`, ... I took the liberty of pushing a commit fixing this.